src/pages/studio/create_post.tsx

```diff
@@ -93,8 +92,40 @@ const CreatePost: React.FC = () => {
 
   const { isOpen, onOpen, onClose } = useDisclosure();
   const router = useRouter();
+
+  const [authenticated, setAuthenticated] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    const token = localStorage.getItem("access_token"); // Retrieve the token from your storage or context
+    if (token) {
+      api.auth.checkAuthentication(token).then((isAuthenticated) => {
+        setAuthenticated(isAuthenticated);
+      });
+    } else {
+      setAuthenticated(false);
+    }
+  }, []);
```

Comment:
suggestion/feature request: We should move ALL this logic here into a Zustand hook so it works globally across the entire frontend. At the moment this essentially creates a scenario where we have potentially multiple requests being triggered across the codebase for the exact same thing.

This would be accomplished through the following:
- Zustand hook for managing state
- Global context to trigger the initial check and only once across the codebase